### PR TITLE
[PM-13] Add new exercises

### PR DESCRIPTION
**Pull request by brian-norman**
Decided to go with full screen dialog as per the [Material Design Guidelines](https://material.io/components/dialogs/#full-screen-dialog). I'm also not doing error checking for input yet, so it'll crash if you don't enter valid data! Felt like a good PR to get out regardless to unblock the squad 😄

### CODE REVIEW

File: app/src/main/java/io/mochahub/powermeter/models/Exercise.kt [hunk 1]
```diff
@@ -2,7 +2,7 @@ package io.mochahub.powermeter.models
 
 data class Exercise(
     val name: String,
-    val personalRecord: Float,
+    val personalRecord: Double,
```
Comment (brian-norman) [inline app/src/main/java/io/mochahub/powermeter/models/Exercise.kt:5]: We used Double in the Room Entity but didn't match it here!

File: app/src/main/res/menu/menu_new_exercise.xml [hunk 2]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="utf-8"?>
+<menu xmlns:android="http://schemas.android.com/apk/res/android"
+        xmlns:app="http://schemas.android.com/apk/res-auto">
+    <item
+            android:id="@+id/action_save"
+            android:title="SAVE"
```
Comment (zahin-mohammad) [inline app/src/main/res/menu/menu_new_exercise.xml]: Can we make this a string resource?

File: app/src/main/res/layout/dialog_new_exercise.xml [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+                android:layout_height="wrap_content" />
+
+    </com.google.android.material.textfield.TextInputLayout>
+
+    <com.google.android.material.textfield.TextInputLayout
+            android:id="@+id/newExerciseGroup"
+            style="@style/Widget.MaterialComponents.TextInputLayout.OutlinedBox"
+            android:layout_width="match_parent"
+            android:layout_height="wrap_content"
+            android:hint="Muscle Group"
```
Comment (zahin-mohammad) [inline app/src/main/res/layout/dialog_new_exercise.xml]: Can we make this a string resource?

File: app/src/main/res/layout/dialog_new_exercise.xml [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+                android:layout_height="wrap_content" />
+
+    </com.google.android.material.textfield.TextInputLayout>
+
+    <com.google.android.material.textfield.TextInputLayout
+            android:id="@+id/newExercisePR"
+            style="@style/Widget.MaterialComponents.TextInputLayout.OutlinedBox"
+            android:layout_width="match_parent"
+            android:layout_height="wrap_content"
+            android:hint="Personal Record"
```
Comment (zahin-mohammad) [inline app/src/main/res/layout/dialog_new_exercise.xml]: Can we make this a string resource?

File: app/src/main/res/layout/dialog_new_exercise.xml [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+            app:layout_constraintTop_toTopOf="parent"
+            app:navigationIcon="@drawable/ic_close_black_24dp"
+            app:theme="@style/ToolBarStyle" />
+
+    <com.google.android.material.textfield.TextInputLayout
+            android:id="@+id/newExerciseName"
+            style="@style/Widget.MaterialComponents.TextInputLayout.OutlinedBox"
+            android:layout_width="match_parent"
+            android:layout_height="wrap_content"
+            android:hint="Name"
```
Comment (zahin-mohammad) [inline app/src/main/res/layout/dialog_new_exercise.xml]: Can we make this a string resource?

File: app/src/main/java/io/mochahub/powermeter/exercises/NewExerciseDialog.kt [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+        }
+    }
+
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View? {
+        super.onCreateView(inflater, container, savedInstanceState)
+        // TODO: The toolbar's X icon isn't final, need to figure out how to make it custom to night mode
```
Comment (zahin-mohammad) [inline app/src/main/java/io/mochahub/powermeter/exercises/NewExerciseDialog.kt]: Don't think it needs to be custom to night mode. It should always stay white I think, like the toolbar title.

File: app/src/main/res/drawable/ic_close_black_24dp.xml [hunk 7]
```diff
@@ -0,0 +1,9 @@
+<vector xmlns:android="http://schemas.android.com/apk/res/android"
+        android:width="24dp"
+        android:height="24dp"
+        android:viewportWidth="24.0"
+        android:viewportHeight="24.0">
+    <path
+        android:fillColor="#FF000000"
+        android:pathData="M19,6.41L17.59,5 12,10.59 6.41,5 5,6.41 10.59,12 5,17.59 6.41,19 12,13.41 17.59,19 19,17.59 13.41,12z"/>
+</vector>
```
Comment (zahin-mohammad) [inline app/src/main/res/drawable/ic_close_black_24dp.xml]: Let's make this white and rename to `ic_close_white_24dp.xml`

**Review by zahin-mohammad — CHANGES_REQUESTED** [review]
[Bug] New exercise gets duped in UI.
- Create a new exercise
- Navigate to another page via navbar
- Navigate to exercises page
- Observe dupe

**Comment by brian-norman** [comment]
@zahin-mohammad nice catch with the bug! Thanks 💯

**Comment by zahin-mohammad** [comment]
[PM-32] (for tracking)

[PM-32]: https://mochahub.atlassian.net/browse/PM-32

File: app/src/main/res/layout/dialog_new_exercise.xml [hunk 8]
```diff
@@ -0,0 +1,70 @@
...
+            app:navigationIcon="@drawable/ic_close_white_24dp"
+            app:theme="@style/ToolBarStyle" />
+
+    <com.google.android.material.textfield.TextInputLayout
+            android:id="@+id/newExerciseName"
+            style="@style/Widget.MaterialComponents.TextInputLayout.OutlinedBox"
+            android:layout_width="match_parent"
+            android:layout_height="wrap_content"
+            android:hint="@string/name"
+            android:padding="16dp"
```
Comment (zahin-mohammad) [inline app/src/main/res/layout/dialog_new_exercise.xml]: Padding will cause the hint and the textbox outline to overlap when typing. Margin will give the expected result.
```suggestion
            android:layout_margin="16dp"
```

File: app/src/main/res/layout/dialog_new_exercise.xml [hunk 9]
```diff
@@ -0,0 +1,70 @@
...
+
+    </com.google.android.material.textfield.TextInputLayout>
+
+    <com.google.android.material.textfield.TextInputLayout
+            android:id="@+id/newExercisePR"
+            style="@style/Widget.MaterialComponents.TextInputLayout.OutlinedBox"
+            android:layout_width="match_parent"
+            android:layout_height="wrap_content"
+            android:hint="@string/personal_record"
+            android:padding="16dp"
```
Comment (zahin-mohammad) [inline app/src/main/res/layout/dialog_new_exercise.xml]: Padding will cause the hint and the textbox outline to overlap. Margin will give the expected result.
```suggestion
            android:layout_margin="16dp"
```

File: app/src/main/res/layout/dialog_new_exercise.xml [hunk 10]
```diff
@@ -0,0 +1,70 @@
...
+
+    </com.google.android.material.textfield.TextInputLayout>
+
+    <com.google.android.material.textfield.TextInputLayout
+            android:id="@+id/newExerciseGroup"
+            style="@style/Widget.MaterialComponents.TextInputLayout.OutlinedBox"
+            android:layout_width="match_parent"
+            android:layout_height="wrap_content"
+            android:hint="@string/muscle_group"
+            android:padding="16dp"
```
Comment (zahin-mohammad) [inline app/src/main/res/layout/dialog_new_exercise.xml]: Padding will cause the hint and the textbox outline to overlap. Margin will give the expected result.
```suggestion
            android:layout_margin="16dp"
```

File: app/src/main/res/layout/dialog_new_exercise.xml [hunk 11]
```diff
@@ -0,0 +1,70 @@
...
+            android:layout_width="match_parent"
+            android:layout_height="wrap_content"
+            android:hint="@string/personal_record"
+            android:padding="16dp"
+            app:layout_constraintEnd_toEndOf="parent"
+            app:layout_constraintStart_toStartOf="parent"
+            app:layout_constraintTop_toBottomOf="@id/newExerciseName">
+
+        <com.google.android.material.textfield.TextInputEditText
+                android:id="@+id/newExercisePRText"
```
Comment (zahin-mohammad) [inline app/src/main/res/layout/dialog_new_exercise.xml:46]:
```suggestion
                android:id="@+id/newExercisePRText"
                android:inputType="numberDecimal"
```

**Review by zahin-mohammad — CHANGES_REQUESTED** [review]
I added some code suggestions.

**Review by zahin-mohammad — APPROVED** [review]

**Pull request merged**